core/src/main/java/org/mule/runtime/core/api/construct/Pipeline.java

```diff
@@ -28,7 +29,9 @@
 
   public List<Processor> getMessageProcessors();
 
-  public void setProcessingStrategy(ProcessingStrategy processingStrategy);
+  public void setProcessingStrategyFactory(ProcessingStrategyFactory processingStrategyFactory);
+
+  public ProcessingStrategyFactory getProcessingStrategyFactory();
```

Comment:
Is this public getter really needed?  Surely the processing strategy shouldn't be used outside of Pipeline, and even if it is then the ProcessingStrategy should be exposed and not the factory.